lib/Dialect/HW/HWOps.cpp

```diff
@@ -2046,6 +2047,13 @@ void StructExplodeOp::print(OpAsmPrinter &printer) {
   printer << " : " << getInput().getType();
 }
 
+void StructExplodeOp::getAsmResultNames(
+    function_ref<void(Value, StringRef)> setNameFn) {
+  auto structType = type_cast<StructType>(getInput().getType());
+  for (auto [res, field] : llvm::zip(getResults(), structType.getElements()))
+    setNameFn(res, field.name.str());
```

Comment:
Is there some way to account for the asm name of the struct? Like

```
%s = hw.struct_create ...
%s_data = hw.struct_extract["data"]
```

Reply:
This was initially the goal of this work! I tried (creating an AsmState and using `Value::printAsOperand`), but this unfortunately bugs up the `AsmState`. My take is that the asm results of a value should be independent of the surrounding IR, much like how operand and result types must be.